crates/bin/pcli/src/transaction_view_ext.rs

```diff
@@ -0,0 +1,349 @@
+                            let bytes = spend.body.nullifier.to_bytes(); // taken to be a unique value, for aesthetic reasons
+                            action = format_opaque_bytes(&bytes);
+                            ["Spend", &action]
+                        }
+                    }
+                }
+                penumbra_transaction::ActionView::Output(output) => {
+                    match output {
+                        OutputView::Visible {
+                            output: _,
+                            note,
+                            payload_key: _,
+                        } => {
+                            action = format!(
+                                "{} -> {}",
+                                format_value_view(&note.value),
+                                format_address_view(&note.address),
+                            );
+                            ["Output", &action]
+                        }
+                        OutputView::Opaque { output } => {
+                            let bytes = output.body.note_payload.encrypted_note.0; // taken to be a unique value, for aesthetic reasons
+                            action = format_opaque_bytes(&bytes);
+                            ["Output", &action]
+                        }
+                    }
+                }
+                penumbra_transaction::ActionView::Swap(swap) => {
+                    // Typical swaps are one asset for another, but we can't know that for sure.
+                    match swap {
+                        SwapView::Visible {
+                            swap: _,
+                            swap_plaintext,
+                        } => {
+                            let (from_asset, from_value, to_asset) = match (
+                                swap_plaintext.delta_1_i.value(),
+                                swap_plaintext.delta_2_i.value(),
+                            ) {
+                                (0, v) if v > 0 => (
+                                    swap_plaintext.trading_pair.asset_2(),
+                                    swap_plaintext.delta_2_i,
+                                    swap_plaintext.trading_pair.asset_1(),
+                                ),
+                                (v, 0) if v > 0 => (
+                                    swap_plaintext.trading_pair.asset_1(),
+                                    swap_plaintext.delta_1_i,
+                                    swap_plaintext.trading_pair.asset_2(),
+                                ),
+                                // The pathological case (both assets have output values).
+                                _ => (
+                                    swap_plaintext.trading_pair.asset_1(),
+                                    swap_plaintext.delta_1_i,
+                                    swap_plaintext.trading_pair.asset_1(),
+                                )
+                            };
```

Comment:
Similarly to issues displaying Fee - does TradingPairView exist?

Reply:
Nope -- so doing something simple is good, and we can collect a list of missing `View`s to add. Nice catch!